src/service/history-impl.js

```diff
@@ -0,0 +1,659 @@
+/**
```

Comment:
Correction. It does auto detect the copy, but apparently github doesn't show it.